crates/dekaf/src/topology.rs

```diff
@@ -87,7 +89,16 @@ impl Collection {
         } else {
             &spec.read_schema_json
         };
-        let (key_schema, value_schema) = avro::json_schema_to_avro(json_schema, &key_ptr)?;
+
+        let json_schema = doc::validation::build_bundle(json_schema)?;
+        let validator = doc::Validator::new(json_schema)?;
+        let mut shape = doc::Shape::infer(&validator.schemas()[0], validator.schema_index());
+
+        if matches!(deletion_mode, DeletionMode::CDC) {
+            shape.widen(&serde_json::json!({"_meta":{"is_deleted":1}}));
```

Comment:
@jgraettinger does this seem like the best way to go about adding a new field to the schema? It was the simplest way I found, but do you think that somehow directly modifying the `Shape` would be better?

Reply:
seems reasonable to me